packages/micropip/src/micropip/_micropip.py

```diff
@@ -312,10 +315,43 @@ async def add_wheel(self, name, wheel, version, extras, ctx, transaction):
 
         with ZipFile(io.BytesIO(wheel_bytes)) as zip_file:
             dist = pkg_resources_distribution_for_wheel(zip_file, name, "???")
+
+        # the package name and version in the wheel file name can be different from
+        # real package project name specified in METADATA file.
+        # The most common example of this is when the package name contains "-",
+        # it will be replaced with "_" in wheel file name.
+        real_name = dist.project_name
+        if real_name == "UNKNOWN":
+            real_name = name
+        try:
+            real_version = dist.version
+        except ValueError:
+            real_version = version
+
+        if real_name != name or real_version != version:
+            transaction["locked"].pop(name)
+            transaction["locked"][real_name] = PackageMetadata(
```

Comment:
I'm not sure that's the best approach. 

How about rather in the list of our locked and installed packages use `name.replace('-', '_')` as key? Then updating the actual name in the PackageMetadata.name.
That would be better than erroring if multiple variants are installed. Unless there are cases when a package name with a '-' vs '_' are different packages, but I think that would very ill-advised anyway.

Reply:
Yeah +1 for this. We should be normalizing the keys. Maybe packaging has a method to do that?

Reply:
Yes, I think keeping both the normalized name and actual project name will work without this update thing. I'll try with that. Thanks for the suggestion.